src/app/clusters/commodity-metering-server/commodity-metering-server.cpp

```diff
@@ -0,0 +1,360 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "commodity-metering-server.h"
+
+#include <protocols/interaction_model/StatusCode.h>
+
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/EventLogging.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/support/CodeUtils.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Globals;
+using namespace chip::app::Clusters::Globals::Structs;
+using namespace chip::app::Clusters::CommodityMetering;
+using namespace chip::app::Clusters::CommodityMetering::Attributes;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommodityMetering {
+
+// Some constraints for lists limitation
+constexpr uint8_t kMaxMeteredQuantityEntries                    = 128;
```

Comment:
Where is this coming from?  I don't see it in the spec, as of spec SHA 9da9b504fc434f25d35f87f8d108eb19043416ae.